<?xml version="1.0" encoding="UTF-8"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro"  name="camtest">

    <xacro:include filename="robot_inertia.xacro" />

    <gazebo>
        <static>true</static>
    </gazebo>

    <link name="world">
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </link>

    <link name="rgb_link">
        <origin xyz="0 0 0.015" rpy="0 0 0"/>
        <visual>
            <geometry>
                <box size="0.03 0.03 0.03"/>
            </geometry>
        </visual>
        <collision>
            <geometry>
                <box size="0.03 0.03 0.03"/>
            </geometry>
        </collision>
        <xacro:inertial_box mass="10000.0" x="0.015" y="0.015" z="0.015"/>
        <material name="blue" />
    </link>
 
    <joint name="rgb_joint" type="fixed">
        <parent link="world" />
        <child link="rgb_link" />
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>
 
    <!-- This is for transformations between image frame and world frame -->
    <link name="rgb_link_optical"></link>
 
    <joint name="rgb_optical_joint" type="fixed">
        <parent link="rgb_link" />
        <child link="rgb_link_optical" />
        <origin xyz="0 0 0" rpy="${-pi/2} 0 ${-pi/2}"/>
    </joint>
 
    <gazebo reference="rgb_link">
        <material>Gazebo/Blue</material>
 
        <sensor name="rgb" type="camera">
            <pose> 0 0 0 0 0 0 </pose>
            <visualize>true</visualize>
            <update_rate>10</update_rate>
            <camera>
                <horizontal_fov>1.57079632679</horizontal_fov>
                <image>
                    <format>R8G8B8</format>
                    <width>640</width>
                    <height>480</height>
                </image>
                <clip>
                    <near>0.05</near>
                    <far>8.0</far>
                </clip>
            </camera>
            <plugin name="rgb_controller" filename="libgazebo_ros_camera.so">
                <frame_name>rgb_link_optical</frame_name>
                <ros>
                    <namespace>camera</namespace>
                </ros>
            </plugin>
        </sensor>
    </gazebo>
</robot>
 